clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/cache/AzureLoadBalancerCachingAgent.groovy

```diff
@@ -75,30 +91,223 @@ class AzureLoadBalancerCachingAgent implements CachingAgent, AccountAware {
+    AzureLoadBalancerDescription evictedLoadBalancer = null
+    String loadBalancerName = data.loadBalancerName as String
+    String resourceGroupName = AzureUtilities.getResourceGroupName(AzureUtilities.getAppNameFromAzureResourceName(loadBalancerName), region)
+    if (resourceGroupName == null) {
+      log.info("handle->Unexpected error retrieving resource group name: null")
+      return []
+    }
+
+    try {
+      updatedLoadBalancer = metricsSupport.readData {
+        creds.networkClient.getLoadBalancer(creds, resourceGroupName, loadBalancerName)
+      }
+    } catch (Exception e ) {
+      log.info("handle->Unexpected exception: ${e.message}")
+      return null
+    }
+
+    def cacheResult = metricsSupport.transformData {
+      if (updatedLoadBalancer) {
+        return buildCacheResult(providerCache, null, 0, updatedLoadBalancer, null)
+      } else {
+        evictedLoadBalancer = new AzureLoadBalancerDescription(
+          loadBalancerName: loadBalancerName,
+          region: region,
+          appName: AzureUtilities.getAppNameFromAzureResourceName(loadBalancerName),
+          cloudProvider: "azure",
+          cluster: "none",
+          lastReadTime: System.currentTimeMillis()
+        )
+        return buildCacheResult(providerCache, null, 0, null, evictedLoadBalancer)
+      }
+    }
+    Map<String, Collection<String>> evictions = evictedLoadBalancer ? [(Keys.Namespace.AZURE_LOAD_BALANCERS.ns): [getLoadBalancerKey(evictedLoadBalancer)]] : [:]
+
+    log.info("onDemand cache refresh (data: ${data}, evictions: ${evictions})")
+    return new OnDemandAgent.OnDemandResult(
+      sourceAgentType: getAgentType(), cacheResult: cacheResult, evictions: evictions
+    )
+  }
+
+  @Override
+  Collection<Map> pendingOnDemandRequests(ProviderCache providerCache) {
+    return []
+  }
+
+  @Override
   CacheResult loadData(ProviderCache providerCache) {
     log.info("Describing items in ${agentType}")
+    def currentTime = System.currentTimeMillis()
 
-    def loadBalancers = creds.networkClient.getLoadBalancersAll(creds, region)
-
-    List<CacheData> data = loadBalancers.collect() { AzureLoadBalancerDescription loadBalancer ->
-      Map<String, Object> attributes = [loadbalancer: loadBalancer]
-      new DefaultCacheData(Keys.getLoadBalancerKey(azureCloudProvider,
-                                                   loadBalancer.loadBalancerName,
-                                                   loadBalancer.loadBalancerName,
-                                                   loadBalancer.appName?:"none",
-                                                   loadBalancer.cluster?:"none",
-                                                   region,
-                                                   accountName),
-        attributes,
-        [:])
+    buildCacheResult(providerCache, creds.networkClient.getLoadBalancersAll(creds, region), currentTime, null, null)
+  }
+
+  @WithWriteLock
```

Comment:
Motivation behind this? 

You shouldn't actually have multiple instances of a particular caching agent running simultaineously?